the-collab-lab/tcl-15-smart-shopping-list
# Issue #6 | User avoids entering duplicate items

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |

## Acceptance Criteria


## Description

* The `AddItem` component imports a helper function `existingName` which is responsible for finding duplicate items and for removing punctuation using `regex`
* The alert messages are dynamically displayed with the `displayMessage` function. If the item exists, the user is alerted with a flash error message (in pink). If successfully added, the message is displayed in green.  
* **Output**: The user now avoids entering duplicate items and is alerted when trying to save an existing item to the list. 

## Related Issue


closes #6 

## Updates

* When the user tries to save an existing item to the list, a flash error message is displayed for 5 seconds. 

![Screen Shot 2020-10-20 at 6 48 41 PM](https://user-images.githubusercontent.com/36022831/96618182-e81ddc00-1304-11eb-84cf-a9a67c672fcf.png)

## Testing Steps / QA Criteria

* Pull down this branch with `git pull origin se-dg-avoid-duplicate-items` and `git checkout se-dg-avoid-duplicate-items`
* Run `npm start` to launch the app

* On the browser: 
1. Try adding a new item, check the `list view` and the success message ✨
2. Try adding the same item as before and make sure that it's not saved to the list
3. Finally, check that an error message is displayed
